python_modules/dagster/dagster/grpc/client.py

```diff
@@ -98,17 +99,22 @@ def _channel(self):
             yield channel
 
     def _query(self, method, request_type, timeout=DEFAULT_GRPC_TIMEOUT, **kwargs):
-        with self._channel() as channel:
-            stub = DagsterApiStub(channel)
-            response = getattr(stub, method)(request_type(**kwargs), timeout=timeout)
-        # TODO need error handling here
-        return response
+        try:
+            with self._channel() as channel:
+                stub = DagsterApiStub(channel)
+                response = getattr(stub, method)(request_type(**kwargs), timeout=timeout)
+            return response
+        except Exception as e:
+            raise DagsterUserCodeUnreachableError("Could not reach user code server") from e
```

Comment:
is `_InactiveRpcError` the main exception that's thrown here?

Reply:
I think so, but I don't know enough about the internal gRPC errors to be sure - I kind of want to just treat this as an opaque error boundary as far as 'accessing the user code' is concerned